src/validations/validations.ts

```diff
@@ -1,22 +1,41 @@
+import type { BBox } from 'geojson';
 import { BoundingBox, GeoPoint, type Geometry } from '../classes';
-import { geometryToBoundingBox } from '../converters/geometry';
 import { Tile } from '../tiles/tile';
 import { TileMatrixSet } from '../tiles/tileMatrixSet';
-import { getTileMatrix, tileMatrixToBoundingBox } from '../tiles/tiles';
-import type { TileMatrix } from '../tiles/types';
-import type { ArrayElement, GeoJSONGeometry, TileMatrixId } from '../types';
+import type { TileRange } from '../tiles/tileRange';
+import { tileMatrixToBoundingBox } from '../tiles/tiles';
+import type { TileMatrix, TileMatrixId } from '../tiles/types';
+import type { ArrayElement, GeoJSONGeometry } from '../types';
 
 /**
  * Validates that the input `boundingBox` is valid
  * @param boundingBox the bounding box to validate
  */
-export function validateBoundingBox(boundingBox: BoundingBox): void {
-  if (boundingBox.max.lon <= boundingBox.min.lon) {
-    throw new Error("bounding box's max.lon must be larger than min.lon");
+export function validateBoundingBox(boundingBox: BoundingBox | BBox): void {
```

Comment:
Shouldn't these validations be by the `grid` values? Something like `grid.maxPoint.lon - grid.minPoint.lon` for whole length and compared to `min` and `max` for bounds.

Reply:
do you mean `validateBoundingBoxByTileMatrix`?